op-node/rollup/derive/batches.go

```diff
@@ -25,10 +25,15 @@ const (
 	BatchFuture
 )
 
+func CheckBatchEspresso() BatchValidity {
+	// TODO: verify batch constraints
+	panic("Unimplemented")
+}
+
 // CheckBatch checks if the given batch can be applied on top of the given l2SafeHead, given the contextual L1 blocks the batch was included in.
 // The first entry of the l1Blocks should match the origin of the l2SafeHead. One or more consecutive l1Blocks should be provided.
 // In case of only a single L1 block, the decision whether a batch is valid may have to stay undecided.
-func CheckBatch(cfg *rollup.Config, log log.Logger, l1Blocks []eth.L1BlockRef, l2SafeHead eth.L2BlockRef, batch *BatchWithL1InclusionBlock) BatchValidity {
+func CheckBatch(cfg *rollup.Config, log log.Logger, l1Blocks []eth.L1BlockRef, l2SafeHead eth.L2BlockRef, batch *BatchWithL1InclusionBlock, usingEspresso bool) BatchValidity {
```

Comment:
Would it be possible to just look at the system config from `l2SafeHead` in this function, rather than plumbing the `usingEspresso` flag all the way down? Or we can't because we don't have anything that lets us look up the system config at this point in the pipeline?

Reply:
Could maybe add it to the `BatchData` type then, just so we don't have to pass it around everywhere explicitly. But that's not very important for now